lib/src/database/database_api.dart

```diff
@@ -357,6 +355,16 @@ abstract class DatabaseApi {
 
   Future<DiscoveryInformation?> getWellKnown();
 
+  Future<void> storeDeviceId(String deviceId);
+
+  Future<String?> getDeviceId();
+
+  Future<void> storeOidcAuthMetadata(Map<String, Object?>? authMetadata);
+
+  Future<Map<String, Object?>?> getOidcAuthMetadata();
+
+  Future<void> storeOidcDynamicClientId(String? oidcClientId);
```

Comment:
I'm not sure if this is the right approach to create methods for every single thing we store in the Client Database. This is now already messed up as we use getClient and updateClient to update single fields but now for other single fields we have methods. What about just adding one generic method `getClientData(ClientData key)` and `putClientData(ClientData key, value)` where `ClientData` is an enum of all possible fields